### DISCUSSION TITLE
Apache reverse proxy - any working virtual host scripts available?

### DISCUSSION THREAD

**Discussion by johnvick** [pyload/pyload#3984] [Help]
Python: 2.7.18 (default, May 3 2020, 20:31:45) [GCC 9.2.0]
OS: posix linux2 Linux 35592550dbd0 5.8.0-44-generic #50~20.04.1-Ubuntu SMP 
pyLoad version: 0.4.20
Installation Folder: /app/pyload
Config Folder: /config
Download Folder: /downloads
Free Space:321.35 GiB
Language: en
Webinterface Theme: modern by GammaC0de based on work of Marco Fernandes and Dogan Bagci
Webinterface Port:8007
Remote Interface Port:7227

Using Linuxserver Docker install. The following Apache virtual host semi-works in that the interface appears (but no icons) and any attempt at download gives error - this is with  https://mydns/pyload 

Using ip:port works fine. 

ProxyPass /pyload/ http://192.168.1.2:8007/
ProxyPassReverse /pyload/ http://192.168.1.2:8007/
RewriteRule ^/pyload$ /pyload/ [R]
ProxyHTMLURLMap http://192.168.1.2:8007 /pyload
<Location /pyload/>
    SetOutputFilter proxy-html
    ProxyPassReverse /
    ProxyHTMLURLMap / /pyload/
    RequestHeader unset Accept-Encoding
</Location>

Any better suggestions? Thanks.

**Comment by GammaC0de** [comment]
Try setting `Path Prefix:` config setting (config->general->Webinterface) to /pyload

**Comment by johnvick** [comment]
Thanks but no luck with that.

**Comment by GammaC0de** [comment]
You should use wireshark / tcpdump to investigate what is happening

**Comment by johnvick** [comment]
Thanks that's probably above my level of expertise but I'll read up on it.
I found the reverse proxy info on this French site:
https://www.monlinux.net/2014/10/apache-en-mode-reverse-proxy/
The author seems to get this working but his solution does not work for me.

**Comment by OzzieIsaacs** [comment]
For me this is working:

apache.conf:
```
Redirect /pyload /pyload/
<Location /pyload>
       # for login preserve host has to be set to true in virtual host  
       order allow,deny
       allow from all
       ProxyPass http://localhost:8000
       Header edit Location ^[^/]*(//[^/]*)(.*)$ $1/pyload$2
       Header edit Set-Cookie "^(.*; Path=)(.*)" $1/pyload$2
       RequestHeader unset Accept-Encoding
       SetOutputFilter html;css;js;jsapp
</Location>

ExtFilterDefine html mode=output intype=text/html cmd="/bin/sed s#src=\"/#src=\"/pyload/#g;s#href=\"/#href=\"/pyload/#g;s#/json/#/pyload/json/#g;s#=\"/pathchooser#=\"/pyload/pathchooser#g;s#url=\"/\"#url=\"/pyload\"#g;s#url=\"/dashboard\"#url=\"/pyload\"#g;"
ExtFilterDefine css mode=output intype=text/css cmd="/bin/sed s#/media/#/pyload/media/#g"
ExtFilterDefine js mode=output intype=text/javascript cmd="/bin/sed s#/json/#/pyload/json/#g;s#/media/#/pyload/media/#g;s#/api/#/pyload/api/#g;s#/home#/pyload/home#g;s#/filechooser#/pyload/filechooser#g;s#/pathchooser#/pyload/pathchooser#g"
ExtFilterDefine jsapp mode=output intype=application/javascript cmd="/bin/sed s#/json/#/pyload/json/#g;s#/media/#/pyload/media/#g;s#/api/#/pyload/api/#g;s#/home#/pyload/home#g;s#/filechooser#/pyload/filechooser#g;s#/pathchooser#/pyload/pathchooser#g;s#/dashboard#/pyload/dashboard#g;"

```
An in my sites-file:

```
<VirtualHost *:80>
        ProxyPreserveHost On
        ....
</VirtualHost>
```

I'm having enabled the following mods (not sure if all of them are needed for the above config):
![grafik](https://user-images.githubusercontent.com/6438895/113387170-0bf14d80-938c-11eb-9ecf-2ecee3574776.png)

What it does: If path prefix is not set pyload is not aware of it's location, so the location in the headers of the request are set to the subfolder, furthermore the cookies are set to the subfolder. (in the location block). Furthermore all links in the html, css and js files are changed from e.g. json to pyload/json. So if you are using something different than pyload for your folder you have to change all occurrences of pyload with your foldername (I got this from a very old pyload issue, as far as I remember)

**Reply by sherif-audibene** [reply]
you are genuis, it works
